src/Bynder/Api/Impl/OAuth2/BynderOauthProvider.php

```diff
@@ -22,8 +22,11 @@ class BynderOauthProvider extends AbstractProvider
 
     public function __construct(array $options = [], array $collaborators = [])
     {
-        $this->bynderDomain = 'https://' . $options['bynderDomain'];
-        unset($options['bynderDomain']);
+        if(isset($options['bynderDomain'])) 
```

Comment:
This way down the line it will go and throw an `IllegalArgumentException`. 
But I can see us checking the `domain` and break out with an `exception` here itself if unset. You think that would make more sense ?

Reply:
Should it even be in a structure called "options" then? 🤔  You could also make it a required arg in the constructor.

Reply:
I can see why you would want it as a separate argument which would resolve the concern.

But if we keep it under options we can also resolve this concern. Specifically wrt the exceptions I mean they will be handled but we can be more explicit by throwing out(again, if we keep `bynderDomain` under `options`). 

Nice to see what you guys think is better though!

Reply:
I would say keep the `$options` parameter as is (although we could get rid of the default value) but add a check and throw an error if `bynderDomain` is missing from it. This way people who currently use the SDK with OAuth2 won't have to make changes.